featureFlags/README.md

```diff
@@ -0,0 +1,164 @@
+      'owner': 
+      {
+        'username': string,
+        'img': string
+      },
+      'launched_at': timestamp
+    }
+  ]
+}
+```
+
+- **Error Response:**
+  - **Code:** 500
+    - **Content:** `{ 'statusCode': 500, 'error': 'Internal Server Error', 'message': 'An internal server error occurred' }`
+
+## **POST /featureFlags** 
+
+Creates a feature flag 
+
+- **Params**  
+  None
+- **Query**  
+  None
+- **Headers**  
+  Content-Type: application/json
+- **Cookie**  
+  rds-session: `<JWT>`
+- **Body** `
+{ 
+   _Required:_ 'feature_name'= <unique feature name>,
+   _Required:_ 'feature_title'= <feature title>,
+   _Required:_ 'config'= <Object with one boolean value which specify feature is enabled or disabled>,             
+ }`
+- **Success Response:**
+- **Code:** 200
+  - **Content:**
+
+```
+{
+  message: 'FeatureFlag added successfully!'
+  data: {
+      'name': string,
+      'id': string
+      'title':string,
+      'created_at': timestamp,
+      'updated_at': timestamp,
+      'config': {'enabled': boolean},
+      'owner': string,
+    }
+}
+```
+
+- **Error Response:**
+  - **Code:** 400
+    - **Content:** `{ 'statusCode': 400, 'error': 'Bad Request ', 'message': 'Missing required fields' }`
+  - **Code:** 401
+    - **Content:** `{ 'statusCode': 401, 'error': 'Unauthorized', 'message': 'Unauthenticated User' }`
+  - **Code:** 500
+    - **Content:** `{ 'statusCode': 500, 'error': 'Internal Server Error', 'message': 'An internal server error occurred' }`
+    
+
+## **PATCH /featureFlags/:id**
+
+Updates data of the feature flag 
+
+- **Params**  
+  _Required:_ `id=[string]`
+- **Headers**  
+  Content-Type: application/json
+- **Cookie**  
+  rds-session: `<JWT>`
+- **Body** `{ <featureFlag_object> }`
+- **Success Response:**
+- **Code:** 204
+
+  - **Content:** `<No Content>`
+
+- **Error Response:**
+  - **Code:** 401
+    - **Content:** `{ 'statusCode': 401, 'error': 'Unauthorized', 'message': 'Unauthenticated User' }`
+  - **Code:** 404
+    - **Content** `{ 'statusCode': 404, 'error': 'Not found', 'message': 'No featureFlag found' }`
+  - **Code:** 500
+    - **Content:** `{ 'statusCode': 500, 'error': 'Internal Server Error', 'message': 'An internal server error occurred' }`
+
+## **DELETE /featureFlag/:id**
+
+ Deletes the feature flag 
+
+- **Params**  
+  _Required:_ `id=[string]`
+- **Headers**  
+  Content-Type: application/json
+- **Cookie**  
+  rds-session: `<JWT>`
+- **Success Response:**
+- **Code:** 200
+  - **Content:**
+```
+{
+  message: 'FeatureFlag deleted successfully!'
+}
+```
+- **Error Response:**
+  - **Code:** 401
+    - **Content:** `{ 'statusCode': 401, 'error': 'Unauthorized', 'message': 'Unauthenticated User' }`
+  - **Code:** 404
+    - **Content** `{ 'statusCode': 404, 'error': 'Not found', 'message': 'No featureFlag found' }`
+  - **Code:** 500
+    - **Content:** `{ 'statusCode': 500, 'error': 'Internal Server Error', 'message': 'An internal server error occurred' }`
```

Comment:
I suppose DELETE, PUT and POST will be appOwner/superUser only, so we would be returning 403 if the user has cookie, but is not an appOwner/superUser